tests/pytests/test_multibyte_char_terms.py

```diff
@@ -0,0 +1,1443 @@
+    env.assertEqual(res, expected, message = 'σίγμα')
+
+    # Test suggestions with greek sigma at the end of the word
+    expected = ['νεανίας', 'ΝΕΑΝΊΑΣ', 'Νεανίας']
+    res = conn.execute_command('FT.SUGGET', 'sug', 'ΝΕΑΝΊΑΣ')
+    env.assertEqual(res, expected, message = 'ΝΕΑΝΊΑΣ')
+
+    res = conn.execute_command('FT.SUGGET', 'sug', 'Νεανίας')
+    env.assertEqual(res, expected, message = 'Νεανίας')
+
+    res = conn.execute_command('FT.SUGGET', 'sug', 'νεανίας')
+    env.assertEqual(res, expected, message = 'νεανίας')
+
+    # This is a known limitation, when folding the suggestion, we are comparing
+    # only the first byte of the folded character, and for that reason
+    # 'dreisig' matches 'dreißig' but not 'dreissig'
+    res = conn.execute_command('FT.SUGGET', 'sug', 'dreisig')
+    env.assertEqual(res, ['Dreißig'], message = 'dreisig')
+    res = conn.execute_command('FT.SUGGET', 'sug', 'dreisi')
+    env.assertEqual(res, ['Dreißig'], message = 'dreisi')
+
+    # same case for suggestion with İ characterss which is folded as two
+    # codepoints: (U+0069)(U+0307) : (i)(combining dot above)
+    test_value = 'İ = Letter I with dot above'
+    expected = [test_value]
+    res = conn.execute_command('FT.SUGGET', 'sug', test_value)
+    env.assertEqual(res, expected)
+    # Search with a lowercase i also returns the same suggestion
+    res = conn.execute_command('FT.SUGGET', 'sug', 'i = Letter I with dot above')
+    env.assertEqual(res, expected)
+
+
+def testRussianStemming(env):
+    '''Test stemming with multi-byte character words.'''
+    conn = getConnectionByEnv(env)
+    env.cmd('FT.CREATE', 'idx_stem', 'ON', 'HASH', 'LANGUAGE', 'RUSSIAN',
+            'SCHEMA', 't', 'TEXT')
+
+    env.cmd('FT.CREATE', 'idx_no_stem', 'ON', 'HASH', 'LANGUAGE', 'RUSSIAN',
+            'SCHEMA', 't', 'TEXT', 'NOSTEM')
+
+    # Create sample data
+    # lowercase terms
+    conn.execute_command('HSET', 'doc:1L', 't', 'программирование') # programming
+    conn.execute_command('HSET', 'doc:2L', 't', 'программирования')
+    conn.execute_command('HSET', 'doc:3L', 't', 'программированию')
+    conn.execute_command('HSET', 'doc:4L', 't', 'программированием')
+    conn.execute_command('HSET', 'doc:5L', 't', 'программировании')
+    # mixed case terms
+    conn.execute_command('HSET', 'doc:1M', 't', 'ПРОГРАммирование')
+    conn.execute_command('HSET', 'doc:2M', 't', 'ПРОГРАммирования')
+    conn.execute_command('HSET', 'doc:3M', 't', 'ПРОГРАммированию')
+    conn.execute_command('HSET', 'doc:4M', 't', 'ПРОГРАммированием')
+    conn.execute_command('HSET', 'doc:5M', 't', 'ПРОГРАммировании')
+
+    # Search using the index with stemming
+    expected = [10, 'doc:1M', 'doc:4M', 'doc:5M', 'doc:3M', 'doc:2M',
+                    'doc:1L', 'doc:4L', 'doc:5L', 'doc:3L', 'doc:2L']
+    res = conn.execute_command(
+        'FT.SEARCH', 'idx_stem', 'программирование', 'NOCONTENT',
+        'LANGUAGE', 'RUSSIAN', 'SORTBY', 't')
```

Comment:
In original [PR](https://github.com/RediSearch/RediSearch/pull/5391/files#diff-3b6918e2e9b8e379d9a42267280b3a98eaec62d5fdf76e824f17fa4ba6696a91R1300), there is no `LANGUAGE` specified in the query - it has no effect since index is defined with the same LANGUAGE, right?

```
res = conn.execute_command(
        'FT.SEARCH', 'idx_stem', 'программирование', 'NOCONTENT',
        'SORTBY', 't')
```

Reply:
Yes. It's in the PR description